chainio/clients/elcontracts/reader.go

```diff
@@ -116,7 +118,34 @@ func BuildELChainReader(
 	), nil
 }
 
+func BuildELChainReaderFromConfig(
+	cfg types.ElChainReaderConfig,
+	ethClient eth.Client,
+	logger logging.Logger,
+) (*ELChainReader, error) {
```

Comment:
this is literally the same fct as `BuildELChainReader` above, except it takes a cfg instead of the two fcts as arguments. There's literally no difference. I think if you want the cfg to have optional fields you need to take a pointer to the cfg right?

In any case both should behave the same (aka if we go this route, which I'm not super convinced by), then BuildELChainReader should also check whether the addresses are set or not

Reply:
the difference here is
- In the config nothing is required, you can pass an empty and we won't initialize relevant things.
- Now we can add new contract addresses like rewards coordinator without breaking any method signature.

Reply:
I am passing the cfg directly into `chainioutils.NewEigenLayerContractBindingsFromConfig` which is checking if contract address is empty or not and initialized based on thaat

Reply:
ok 100% agree. Let's change name to just `reader.BuildFromConfig` to remoe redundancy. We need to change this across the board as well.

Reply:
Done